From a396908d25976c030e7a6a56bd5be0738c11d2ee Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Sun, 28 Jun 2020 19:44:17 +0200
Subject: [PATCH 28/51] Fix invalid "Regex" Type in various Condition blocks

Regex is not a valid Condition type, using it leads to errors like these:

ALSA lib ucm_cond.c:300:(if_eval) unknown If.Condition.Type
ALSA lib main.c:983:(snd_use_case_mgr_open) error: failed to import cht-bsw-rt5672 use case configuration -22
alsaucm: error failed to open sound card cht-bsw-rt5672: Invalid argument

Replacing it with "RegexMatch" fixes this.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 ucm2/cht-bsw-rt5672/HiFi.conf         | 4 ++--
 ucm2/chtnau8824/HiFi.conf             | 2 +-
 ucm2/chtrt5645/HiFi.conf              | 6 +++---
 ucm2/sof-soundwire/sof-soundwire.conf | 4 ++--
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/ucm2/cht-bsw-rt5672/HiFi.conf b/ucm2/cht-bsw-rt5672/HiFi.conf
index 394e45d39ace..dc8ddd5ed45f 100644
--- a/ucm2/cht-bsw-rt5672/HiFi.conf
+++ b/ucm2/cht-bsw-rt5672/HiFi.conf
@@ -9,7 +9,7 @@ Define.HeadsetMic "yes"
 
 If.cfg-dmic1 {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		String "${CardLongName}"
 		Regex "(SoMeThInK1)" # fixme!
 	}
@@ -21,7 +21,7 @@ If.cfg-dmic1 {
 
 If.cfg-dmic2 {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		String "${CardLongName}"
 		Regex "(LENOVO.*ThinkPad8)"
 	}
diff --git a/ucm2/chtnau8824/HiFi.conf b/ucm2/chtnau8824/HiFi.conf
index 6b7624d6e775..3ccd6b149517 100644
--- a/ucm2/chtnau8824/HiFi.conf
+++ b/ucm2/chtnau8824/HiFi.conf
@@ -2,7 +2,7 @@ Define.Speaker "Speaker"
 
 If.cfg-mspk {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		String "${CardLongName}"
 		Regex "(cube-i1_TF|PIPO-W2S)"
 	}
diff --git a/ucm2/chtrt5645/HiFi.conf b/ucm2/chtrt5645/HiFi.conf
index fb8636d8ffac..5d0e3bf42f40 100644
--- a/ucm2/chtrt5645/HiFi.conf
+++ b/ucm2/chtrt5645/HiFi.conf
@@ -4,7 +4,7 @@ Define.DigitalMic ""
 
 If.cfg-dmic1 {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		String "${CardLongName}"
 		Regex "(ASUSTeK.*T100HAN|ASUSTeK.*T101HA)"
 	}
@@ -16,7 +16,7 @@ If.cfg-dmic1 {
 
 If.cfg-dmic2 {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		String "${CardLongName}"
 		Regex "(LENOVO.*LenovoMIIX320|MEDION.*Wingman)"
 	}
@@ -28,7 +28,7 @@ If.cfg-dmic2 {
 
 If.cfg-mspk {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		String "${CardLongName}"
 		Regex "(gpd-win-pocket-rt5645|TECLAST-X80Pro)"
 	}
diff --git a/ucm2/sof-soundwire/sof-soundwire.conf b/ucm2/sof-soundwire/sof-soundwire.conf
index fe43d5f81d57..2ba44be8b284 100644
--- a/ucm2/sof-soundwire/sof-soundwire.conf
+++ b/ucm2/sof-soundwire/sof-soundwire.conf
@@ -38,7 +38,7 @@ DefineRegex {
 
 If.hs_init {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		Regex "(rt5682|rt700|rt711)"
 		String "${var:HeadsetCodec1}"
 	}
@@ -47,7 +47,7 @@ If.hs_init {
 
 If.mic_init {
 	Condition {
-		Type Regex
+		Type RegexMatch
 		Regex "(rt715)"
 		String "${var:MicCodec1}"
 	}
-- 
2.16.4

